using-timescaledb/custom-jobs.md

```diff
@@ -0,0 +1,221 @@
+be useful to debug problems when combined with increased log level.
+
+### Altering / Dropping custom jobs [](alter-delete)
+
+You can alter the config or scheduling parameters with `alter_job`.
+
+```sql
+SELECT alter_job(1000, config => '{"hypertable":"metrics"}');
+```
+
+Replaces the config for job with id 1000 with the specified JSON.
+
+```sql
+SELECT alter_job(1000, scheduled => false);
+```
+
+Disables automatic scheduling of the job with id 1000. The job can still be run manually
+with `run_job`.
+
+```sql
+SELECT alter_job(1000, scheduled => true);
+```
+
+Reenables automatic scheduling of the job with id 1000.
+
+```sql
+SELECT delete_job(1000);
+```
+
+Deletes the job with id 1000 from the automation framework.
+
+## Example custom jobs [](examples)
+
+### Generic Retention Policy [](generic-retention-policy)
+
+Retention policy for multiple hypertables. Additional filters can be applied
+by adding a WHERE clause to the query.
+
+```sql
+CREATE OR REPLACE PROCEDURE generic_retention (job_id int, config jsonb)
+LANGUAGE PLPGSQL
+AS $$
+DECLARE
+  lag interval;
+BEGIN
+  SELECT jsonb_object_field_text (config, 'lag')::interval INTO STRICT lag;
+
+  IF lag IS NULL THEN
+    RAISE EXCEPTION 'Config must have lag';
+  END IF;
+
+  PERFORM drop_chunks(format('%I.%I', table_schema, table_name), older_than => lag)
+    FROM timescaledb_information.hypertables;
+END
+$$;
+```
+
+Register job to run daily dropping chunks on all hypertables that are older
+than 12 months.
+
+```sql
+SELECT add_job('generic_retention','1d', config => '{"lag":"12 month"}');
+```
+
+### Tiered storage [](tiered-storage)
+
+Custom job that moves chunks older than a certain time to a different tablespace.
+
+```sql
+CREATE OR REPLACE PROCEDURE move_chunks (job_id int, config jsonb)
+LANGUAGE PLPGSQL
+AS $$
+DECLARE
+  ht REGCLASS;
+  lag interval;
+  destination name;
+  chunk REGCLASS;
+  tmp_name name;
+BEGIN
+  SELECT jsonb_object_field_text (config, 'hypertable')::regclass INTO STRICT ht;
+  SELECT jsonb_object_field_text (config, 'lag')::interval INTO STRICT lag;
+  SELECT jsonb_object_field_text (config, 'tablespace') INTO STRICT destination;
+
+  IF ht IS NULL OR lag IS NULL OR destination IS NULL THEN
+    RAISE EXCEPTION 'Config must have hypertable, lag and destination';
+  END IF;
+
+  FOR chunk IN
+  SELECT show.oid
+  FROM show_chunks(ht, older_than => lag)
+  SHOW (oid)
+    INNER JOIN pg_class pgc ON pgc.oid = show.oid
+    INNER JOIN pg_tablespace pgts ON pgts.oid = pgc.reltablespace
+  WHERE pgts.spcname != destination;
+  LOOP
+    RAISE NOTICE 'Moving chunk: %', chunk::text;
+    EXECUTE format('ALTER TABLE %s SET TABLESPACE %I;', chunk, destination);
+  END LOOP;
+END
+$$;
+```
+
+Register job to run daily moving chunks older than 12 months on hypertable
+metrics to tablespace old_chunks.
+
+```sql
+SELECT add_job('move_chunks','1d', config => '{"hypertable":"metrics","lag":"12 month","tablespace":"old_chunks"}');
+```
+
```

Comment:
Maybe a note like the following at bottom of section:

```
The above action uses the simpler `ALTER TABLE ... SET TABLESPACE` for moving
a chunk, although it could alternatively be written in terms of TimescaleDB's
[`move_chunk`](/api#move_chunk).  The `move_chunk` function also requires an
index as input, but performs data re-ordering as part of the move (for faster
subsequent queries) and requires lower lock levels, so the chunk remains available
for reads during the move.
```